lib/net.js

```diff
@@ -397,6 +397,14 @@ Socket.prototype.setTimeout = function(msecs, callback) {
 
 
 Socket.prototype._onTimeout = function() {
+  // `.prevWriteQueueSize` !== `.updateWriteQueueSize()` means there is
+  // an active write in progress, so we suppress the timeout.
+  const prevWriteQueueSize = this._handle.writeQueueSize;
+  if (prevWriteQueueSize > 0 &&
+      prevWriteQueueSize !== this._handle.updateWriteQueueSize()) {
+    this._unrefTimer();
+    return;
```

Comment:
Why unref? That doesn't quite match what I understood when I read "suppress the timeout"?

Reply:
It'll call `_unrefActive` so that it can come back and check this condition again in whatever the `_idleTimeout` is, if need be. Possible the comment is confusing but I'm obviously familiar with what it does so I'm a bad judge.

Reply:
So … I might be wrong here, but judging from the naming I’d think `_unrefActive` only affects whether the timer handle is ref()ed or not, i.e. whether it is keeping the process alive or not … does that sound about right?

Reply:
Sorry, missed the notification — `_unrefActive` re-schedules a timer for a timer that's a part of the `unrefedLists`.

https://github.com/nodejs/node/blob/686e092202514dfc3cf93a463e9177a80dd42133/lib/timers.js#L141-L151

Not sure if that's a great explanation... ? Let me know if it isn't.

Reply:
Yeah, don’t feel like you need to wait on anything from me! There’s nothing obviously wrong in this PR.

And yes, this should just be renamed…

Reply:
👍 Just being extra safe since this will go into LTS and is a fairly major change to something that's been broken for quite a while. :)

Reply:
Is there a specific time frame? You can just add a `baking-for-lts` label to indicate that the releaser (likely @gibfahn) should let this sit a while before pulling it in, *if* you would feel more comfortable with that.

(On the other hand, if you feel comfortable not waiting any longer than usual, feel free to voice that as well!)

Reply:
Since it's a reasonably serious bug fix, it hasn't broken CITGM and no one else has expressed desire for having it bake, I think having it go in at the next opportunity should be fine. Thanks again for reviewing ❤️